### DISCUSSION TITLE
How can use DSImageVM inside for loop

### DISCUSSION THREAD

**Discussion by tumi91** [imodeveloperlab/dskit#6] [Q&A]
How can access DSImageVM from outside the loop to assign to the DSPageControlVM that has array params which take DSImageVM value?

// this an example
extension Home2ViewController {
    
    /// Products gallery
    /// - Returns: DSSection
    func productsGallery() -> DSSection {
        
        
        let url = URL(string: "https://example/api/v1/getPost")!

        let task = URLSession.shared.dataTask(with: url) {(data, response, error) in
            guard let data = data else { return }
            print(String(data: data, encoding: .utf8)!)
            
            for each (product in data)
            {
                 let sneakers =  DSImageVM(imageUrl: URL(string:  data.image ), height: .absolute(200))
            }
        
        }
        
        task.resume()
        
        
      
   
        let pageControl = DSPageControlVM(type: .viewModels([sneakers]))

        return pageControl.list().zeroLeftRightInset()
                
            
            
        }

        
    
}

**Comment by imodeveloper** [comment]
Hey, you need separately process request making and UI

1. make the request
2. store data somewhere in your controller, like `var arrayOfProducts: ObjectClass`
3. call a function that will use your data and update your UI after.

**Reply by imodeveloper** [reply]
Also you can explore this project, here you will see real examples with async data https://github.com/imodeveloperlab/RSS-Feed

**Reply by imodeveloper** [reply]
Let me know if this didn't help, I will make a working demo.
